src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,257 @@
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.meeting.Meeting;
+import seedu.address.model.person.Person;
+
+/**
+ * Edits the details of an existing person in the address book.
+ */
+public class EditMeetingCommand extends Command {
+
+    public static final String COMMAND_WORD = "editMeeting";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the meeting identified "
+            + "by the index number used in the displayed meeting list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_CLIENT_INDEX + "CLIENT INDEX] "
+            + "[" + PREFIX_MEETING_INDEX + "MEETING INDEX] "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_DATETIME + "DATETIME] \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_CLIENT_INDEX + "1 "
+            + PREFIX_MEETING_INDEX + "2 "
+            + PREFIX_NAME + "starbucks meeting "
+            + PREFIX_DATETIME + "01-01-2024 12:00 ";
+
+    public static final String MESSAGE_EDIT_MEETING_SUCCESS = "Edited Meeting: %1$s";
+    public static final String MEETING_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_MEETING = "This meeting already exists in the address book.";
+
+    private final Index meetingIndex;
+    private final Index clientIndex;
+
+    private final EditMeetingDescriptor editMeetingDescriptor;
+
+    /**
+     * @param clientIndex of the person in the filtered person list to edit
+     * @param meetingIndex of the meeting in the filtered meeting list to edit
+     * @param editPersonDescriptor details to edit the person with
+     */
+    public EditMeetingCommand(Index clientIndex, EditMeetingDescriptor editPersonDescriptor, Index meetingIndex) {
+        requireNonNull(meetingIndex);
+        requireNonNull(editPersonDescriptor);
+        requireNonNull(clientIndex);
+
+        this.meetingIndex = meetingIndex;
+        this.editMeetingDescriptor = new EditMeetingDescriptor(editPersonDescriptor);
+        this.clientIndex = clientIndex;
+
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+        /*
+        lastShownList.add(new Meeting("a", LocalDateTime.now(), selectedClient));
+        lastShownList.add(new Meeting("b", LocalDateTime.now(), selectedClient));
+        lastShownList.add(new Meeting("c", LocalDateTime.now(), selectedClient));
+        */
```

Comment:
Is there a reason why you left the commented out code in?

Reply:
yea i was trying to test the execution before the adding meeting command was added, i can remove it

Reply:
Alright, can just clean it up when you're free!